File: 69. Sqrt(x)/mySqrt.java
class mySqrt {
    public int solution(int x) {
        long left = 0, right = (x / 2) + 1;
        if (x == 0) {
            return 0;
        }
        while (left < right) {
            long mid = left + (right - left + 1) / 2;
            long square = mid * mid;
            if (square > x) {
                right = mid - 1;
            }
            else {
                left = mid;
            }
        }
        return (int) left;
    }
}

File: 69. Sqrt(x)/mySqrt.py
class mySqrt:
    def solution(self, x: int) -> int:
        if x == 0:
            return 0
        left = 0
        right = x // 2 + 1
        while left < right:
            mid = left + (right - left + 1) // 2
            if mid * mid > x:
                right = mid - 1
            else:
                left = mid
        return int(left)